workflows/iuf/operations/management-nodes-rollout/management-worker-nodes-reboot.yaml

```
@@ -0,0 +1,547 @@
+                      labeled_workers=$(kubectl get nodes --selector='iuf-prevent-reboot=true' -o jsonpath='{range .items[*]}{@.metadata.name}{" "}')
+                      for labeled_worker in $labeled_workers; do
+                        worker_nodes_to_reboot=$(echo $worker_nodes_to_reboot | sed s/"$labeled_worker"//)
+                      done
+                    fi
+
+                    index=0
+                    set=""
+                    output=""
+                    for node in $worker_nodes_to_reboot; do
+                      index=$(( index + 1 ))
+                      set="${set},${node}"
+                      if [[ $index -eq $workers_per_set ]]; then
+                        output="${output} ${set#?}"
+                        index=0
+                        set=""
+                      fi
+                    done
+                    if [[ -n ${set} ]]; then
+                      output="${output} ${set#?}"
+                    fi
+                    echo "${output#?}"
+          - name: process-worker-reboot-sets
+            dependencies:
+              - get-worker-reboot-sets
+            template: process-sets
+            arguments:
+              parameters:
+                - name: sets
+                  value: "{{tasks.get-worker-reboot-sets.outputs.result}}"
+          - name: reboot-worker-sets
+            dependencies:
+              - process-worker-reboot-sets
+            template: reboot-set-of-workers
+            arguments:
+              parameters:
+                - name: sets
+                  value: "{{tasks.process-worker-reboot-sets.outputs.parameters.sets}}"
+                - name: counter
+                  value: "0"
+                - name: limit
+                  value: "{{tasks.process-worker-reboot-sets.outputs.parameters.num_sets}}"
+                - name: global_params
+                  value: "{{inputs.parameters.global_params}}"
+          - name: end-operation
+            dependencies:
+              - reboot-worker-sets
+            templateRef:
+              name: workflow-template-record-time-template
+              template: record-time-template
+          - name: prom-metrics
+            dependencies:
+              - start-operation
+              - end-operation
+            template: prom-metrics
+            arguments:
+              parameters:
+              - name: opstart
+                value: "{{tasks.start-operation.outputs.result}}"
+              - name: opend
+                value: "{{tasks.end-operation.outputs.result}}"
+    - name: process-sets
+      inputs:
+        parameters:
+          - name: sets
+      script:
+        image: artifactory.algol60.net/csm-docker/stable/docker.io/alpine/git:2.32.0
+        command: [sh]
+        source: |
+          #!/bin/sh
+          sets="{{inputs.parameters.sets}}"
+          echo $sets > /tmp/sets
+          num_sets=$(echo $sets | wc -w)
+          echo $num_sets > /tmp/num_sets
```

You don't need to change this, but this can be done with an array
 ```suggestion
          sets=({{inputs.parameters.sets}})
          echo ${sets[*]} > /tmp/sets
          echo ${#sets[*]} > /tmp/num_sets
```